### PR TITLE
feat: Update ar_dataset codes applying padding logic

### PR DESCRIPTION
**Pull request by Quasar0311**
pad_token_id == eos_token_id == 2 일 때,

* input_ids
A: [10, 11, 12, 13, 14, 2]
B: [20, 21, 22, 2, 2, 2] ← (여기서 뒤의 2들은 padding 역할)

* attention_mask
A: [1, 1, 1, 1, 1, 1] ← 마지막 eos는 “진짜 eos”
B: [1, 1, 1, 1, 0, 0] ← 뒤의 eos들은 “padding eos”

* labels (padding의 역할을 하는 eos 제외)
A: [..., 14, 2]
B: [..., 2, -100, -100]

* 주의 `return self.tokenizer.encode(s, add_special_tokens=False)`

맨 앞 <bos> 토큰 이용, 데이터 뒷부분에 <eos> 토큰 이용. pad_token_id와 eos_token_id가 같다면 ignore_index (-100)를 사용하여 구분할 수 있도록 함.
config에도 add_bos_token_when_response_start, add_eos_token_when_response_end 제거.

### CODE REVIEW

**Review by ljm565 — APPROVED** [review]
굳 좋습니다
앞으로는 PR 메시지를 아래 형식으로 해주면 감사하겠습니다

```
### Motivation 

### Key changes

### Test

### Notice (optional)
```

File: src/univlt/data_collection/ar_dataset.py [hunk 1]
```diff
@@ -61,82 +65,134 @@ def __init__(
...
-        if self.add_bos:
-            formatted_prompt = self.tokenizer.bos_token + formatted_prompt
-        if self.add_eos:
-            formatted_prompt = formatted_prompt + self.tokenizer.eos_token
+        label = deepcopy(full_prompt_token)
 
-        label = deepcopy(full_prompt_token)        
+        # When pad_token_id == eos_token_id, padding positions use EOS tokens
+        # set to ignore_index so CrossEntropyLoss(ignore_index=ignore_index) excludes them from loss
+        if self.pad_token_id == self.tokenizer.eos_token_id:
```
Comment (ljm565) [inline src/univlt/data_collection/ar_dataset.py]: __init_tokens 함수에서 정의한 self.eos_token_id로 바꾸면 될 것 같은데 맞나요??

Comment (Quasar0311) [reply src/univlt/data_collection/ar_dataset.py]: 아 넵 에러날 뻔했군요 말씀하신 부분으로 다시 커밋했습니다

**Review by ljm565 — APPROVED** [review]
